weaviate/embedded.py

```diff
@@ -49,15 +51,23 @@ def __init__(self, options: EmbeddedOptions):
         self.ensure_paths_exist()
         self.check_supported_platform()
         self._parsed_weaviate_version = ""
-        # regular expression to detect a version number: v[one digit].[1-2 digits].[1-2 digits] - nothing in front or back
-        pattern = re.compile(r"^v\d\.\d{1,2}\.\d{1,2}$")
-        if self.options.version.startswith(GITHUB_RELEASE_DOWNLOAD_URL):
-            # replace with str.removeprefix() after 3.8 has been deprecated
-            self._parsed_weaviate_version = self.options.version[
-                len(GITHUB_RELEASE_DOWNLOAD_URL) :
-            ].split("/")[0]
+        # regular expression to detect a version number: v[one digit].[1-2 digits].[1-2 digits]
+        # optionally there can be a "-rc.[1-2 digits]"
+        # nothing in front or back
+        version_pattern = re.compile(r"^\d\.\d{1,2}\.\d{1,2}?(-rc\.\d{1,2}|$)$")
```

Comment:
I think the regex pattern should be a bit broader still, as we historically have had `alpha` releases in addition to `rc` releases:

https://github.com/weaviate/weaviate/tree/v1.18.0-alpha.1

Reply:
ah, I only checked the release page. Added alpha and beta